cms-derived-data/create_cms_derived_datasets.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+# noqa: E501
+
+import ROOT
+import os
+import subprocess
+import yaml
+import json
+import click
+
+
+"""
+Create CMS derived datasets.
+"""
+
+
+def get_number_events(file_path, data_type):
+    """Return number of events in root file."""
+    myfile = ROOT.TFile.Open(file_path)
+    number_events = 0
+    if data_type == "NanoAODRun1":
+        number_events = myfile.Events.GetEntries()
+    elif data_type == "POET":
+        number_events = myfile.events.GetEntries()
+    return number_events
+
+
+def get_parent_recid(parent_title):
+    """Return parent dataset recid."""
+    cmd = f"cernopendata-client get-metadata --title={parent_title} --output-value recid"
+    recid = subprocess.getoutput(cmd)
+    return recid
+
+
+def get_file_size(file_path):
+    """Return file size."""
+    file_size = os.path.getsize(file_path)/1024.0
+    return file_size
+
+
+def create_record(metadata, data_type):
+    """Create record using the given metadata."""
+    
+    config_file = open("config.yaml", 'r')
+    config = yaml.safe_load(config_file)
+
+    rec = {}
+
+    rec["abstract"] = {}
+    rec["abstract"]["description"] = config["common_values"]["abstract"]["description"]
+   
+    rec["accelerator"] = config["common_values"]["accelerator"]
+    
+    rec["authors"] = config["common_values"]["authors"]
+    
+    rec["collections"] = config["common_values"]["collections"]
+    
+    rec["control_number"] = config["common_values"]["control_number"]
```

Comment:
The "control number" should not be here, it is the persistent identifier which is produced automatically by "minters", and exposed only over REST API. But in the record JSON source files that we produce for storing amongst "fixtures", it should not appear. We should produce only "recid" as input there. And "control_number" will be produced automatically from it during fixtures upload.